pkg/login/social/azuread_oauth.go

```diff
@@ -69,7 +70,7 @@ func (s *SocialAzureAD) UserInfo(client *http.Client, token *oauth2.Token) (*Bas
 		return nil, errors.New("error getting user info: no email found in access token")
 	}
 
-	role := extractRole(claims, s.autoAssignOrgRole)
+	role := extractRole(claims, s.autoAssignOrgRole, s.roleAttributeStrict)
```

Comment:
You should either check if the role is empty and return an error here or return an error from the `extractRole` function and elevate that up if it's not nil.

Reply:
@sivamu Will this be an easy change?

@sakjur / @kminehart is there someone in particular who should also review this?

Reply:
Added the change, apologies for the delay

Reply:
🙏 Thank you, I'll take another look